### ISSUE TITLE
panic with FromBytesWithNulError in codegen

### ISSUE THREAD

**Issue by windy1** [rust-lang/rust-bindgen#2610]
### Input C/C++ Header

I am encountering this when attempting to include the Bonjour SDK for Windows v3.0: https://developer.apple.com/download/all/?q=Bonjour%20SDK%20for%20Windows

### Bindgen Invocation

`build.rs`

```Rust
extern crate bindgen;

use std::env;
use std::path::{Path, PathBuf};

fn main() {
    let mut builder = bindgen::builder().clang_arg("-v");

    println!("DEBUG1");

    if let Ok(val) = env::var("BONJOUR_SDK_HOME") {
        println!("BONJOUR_SDK_HOME = {val}");

        let path = Path::new(&val);
        let lib = path.join("Lib").join("x64");
        let headers = path.join("Include");

        println!("cargo:rustc-link-search={}", lib.display().to_string());
        println!("cargo:rustc-link-lib=dnssd");

        let clang_arg = format!("-I{}", headers.display());
        println!("clang_arg = {clang_arg}");
        builder = builder.clang_arg(clang_arg);
    }

    builder = builder.header("wrapper.h");

    println!("DEBUG2");

    builder
        .generate()
        .expect("failed to generate bindings")
        .write_to_file(PathBuf::from(env::var("OUT_DIR").unwrap()).join("bindings.rs"))
        .expect("failed to write bindings to file");
}
```

`wrapper.h`

```cpp
#include <dns_sd.h>
```

### Actual Results

```
$ RUST_BACKTRACE=1 cargo build
    Blocking waiting for file lock on build directory
   Compiling bonjour-sys v0.1.0 (C:\Users\walke\code\personal\bonjour-sys)
error: failed to run custom build command for `bonjour-sys v0.1.0 (C:\Users\walke\code\personal\bonjour-sys)`
note: To improve backtraces for build dependencies, set the CARGO_PROFILE_DEV_BUILD_OVERRIDE_DEBUG=true environment variable to enable debug information generation.

Caused by:
  process didn't exit successfully: `C:\Users\walke\code\personal\bonjour-sys\target\debug\build\bonjour-sys-a46711db7def5eda\build-script-build` (exit code: 101)  
  --- stdout
  DEBUG1
  BONJOUR_SDK_HOME = C:\Program Files\Bonjour SDK\
  cargo:rustc-link-search=C:\Program Files\Bonjour SDK\Lib\x64
  cargo:rustc-link-lib=dnssd
  clang_arg = -IC:\Program Files\Bonjour SDK\Include
  DEBUG2

  --- stderr
  clang version 16.0.6
  Target: x86_64-pc-windows-msvc
  Thread model: posix
  InstalledDir:
  ignoring nonexistent directory "lib\clang\16\include"
  #include "..." search starts here:
  #include <...> search starts here:
   C:\Program Files\Bonjour SDK\Include
   D:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.37.32822\include
   D:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.37.32822\atlmfc\include
   C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\ucrt
   C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared
   C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\um
   C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\winrt
   C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\cppwinrt
  End of search list.
  thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: FromBytesWithNulError { kind: InteriorNul(1) }', C:\Users\walke\.cargo\registry\src\index.crates.io-6f17d22bba15001f\bindgen-0.66.1\codegen\mod.rs:662:71
  stack backtrace:
     0: std::panicking::begin_panic_handler
               at /rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library\std\src\panicking.rs:593
     1: core::panicking::panic_fmt
               at /rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library\core\src\panicking.rs:67
     2: core::result::unwrap_failed
               at /rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library\core\src\result.rs:1651
     3: core::result::Result<T,E>::unwrap
     4: <bindgen::ir::var::Var as bindgen::codegen::CodeGenerator>::codegen
     5: <bindgen::ir::item::Item as bindgen::codegen::CodeGenerator>::codegen
     6: <bindgen::ir::module::Module as bindgen::codegen::CodeGenerator>::codegen::{{closure}}
     7: <bindgen::ir::module::Module as bindgen::codegen::CodeGenerator>::codegen
     8: <bindgen::ir::item::Item as bindgen::codegen::CodeGenerator>::codegen
     9: bindgen::codegen::codegen::{{closure}}
    10: bindgen::ir::context::BindgenContext::gen
    11: bindgen::codegen::codegen
    12: bindgen::deprecated_target_diagnostic
    13: bindgen::Builder::generate
    14: bindgen::Bindings::write_to_file
    15: core::ops::function::FnOnce::call_once
  note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

### Expected Results

Bindings generated for `dns_sd.h`

**Comment by MolotovCherry** [comment]
I had a panic in 66.1 with very much the same message as yours. It was fixed in 68.1. See if 68.1 also fixes it for you

**Comment by windy1** [comment]
This resolved it for me too, thank you @MolotovCherry!
